packages/mgt/src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -673,12 +690,24 @@ export class MgtPeoplePicker extends MgtTemplatedComponent {
         people = [];
         if (this.type === PersonType.person || this.type === PersonType.any) {
           try {
-            people = (await findPeople(graph, input, this.showMax)) || [];
+            if (this.groupId) {
+              people =
+                (await findUsersFromGroup(
+                  graph,
+                  input,
+                  this.groupId,
+                  this.showMax,
+                  this.type,
```

Comment:
Just an idea, but would passing `PersonType.person` here instead of `this.type` solve the issue of getting duplicate data that we have to filter? This way we ensure only people are fetched in this block and only groups are fetched in the other block.

Reply:
I'm afraid we would lose the contacts in that case. Possible?

Reply:
Very good point, I didn't realize a group could have contacts (and other types) as members. TIL

We might benefit here from reformating the logic by pivoting on groupId instead. What do you think about this:

```js
if (groupId) {
  // findGroupsFromGroup and findUsersFromGroup can be combined in a single function
  // get members of a group by passing type, transitiveSerach, groupType, etc
  people = findGroupMembers(...); 
} else  {
  // same logic as before

  if (type === person or any) {
    people = findPeople(...)
    if (people.lenght < showMax) {
      findUsers(...) 
      ...
    }
  } 

  if (type === group or any && people < showMax) {
    findGroups(...)
    ...
  }
}

```

Reply:
That's actually a great idea. Will be a lot simpler to write and understand!